### PR TITLE
Form editor, frontend only

### PR DESCRIPTION
**Pull request by luontola**
#610

### CODE REVIEW

File: test/clj/rems/test/db/workflow.clj [hunk 1]
```diff
@@ -1,19 +0,0 @@
-(ns rems.test.db.workflow
```
Comment (opqdonut) [inline test/clj/rems/test/db/workflow.clj]: why remove these three test files?

File: src/cljs/rems/administration/components.cljs [hunk 2]
```diff
@@ -0,0 +1,88 @@
+(ns rems.administration.components
+  (:require [clojure.string :as str]
+            [re-frame.core :as rf]))
+
+; reusable form components
```
Comment (opqdonut) [inline src/cljs/rems/administration/components.cljs]: this (and some more info) could be in the ns doc string

File: src/cljs/rems/administration/components.cljs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  (if (number? key)
+    (str key)
+    (name key)))
+
+(defn- keys-to-id [keys]
+  (->> keys
+       (map key-to-id)
+       (str/join "-")))
+
+(defn text-field [context {:keys [keys label placeholder]}]
```
Comment (opqdonut) [inline src/cljs/rems/administration/components.cljs]: please document `context`

File: test/clj/rems/test/db/workflow.clj (hunk 1 above)
Comment (luontola) [reply test/clj/rems/test/db/workflow.clj]: They were duplicating each other. The common functionality has been moved to https://github.com/CSCfi/rems/blob/1828404b0506a86462d3175e83a22cc8bc1e4080/src/clj/rems/db/core.clj#L17 https://github.com/CSCfi/rems/blob/1828404b0506a86462d3175e83a22cc8bc1e4080/test/clj/rems/test/db/core.clj#L36

File: src/clj/rems/api/public.clj [hunk 4]
```diff
@@ -19,7 +19,8 @@
   {:authentication s/Keyword
    (s/optional-key :alternative-login-url) s/Str
    (s/optional-key :extra-pages) [ExtraPage]
-   (s/optional-key :default-language) s/Keyword})
+   (s/optional-key :default-language) s/Keyword
+   (s/optional-key :dev) s/Bool})
```
Comment (opqdonut) [inline src/clj/rems/api/public.clj]: why optional? and why not `:environment :dev`?

**Pull request merged**